### PR TITLE
Increase Select Testing Coverage

### PR DESCRIPTION
**Pull request by JoeButler7**
#### What does this PR do?
Increases testing coverage on `Select`. 
Before the tests were added coverage was at 
| Type       | Fraction | Percentage |
|------------|----------|------------|
| Statements | 188/216  | 87.04%     |
| Branches   | 157/205  | 76.59%     |
| Functions  | 39/42    | 92.86%     |
| Lines      | 177/199  | 88.94%     |

After the new tests were added 
| Type       | Fraction | Percentage |
|------------|----------|------------|
| Statements | 215/216  | 99.54%     |
| Branches   | 182/205  | 88.78%     |
| Functions  | 42/42    | 100%       |
| Lines      | 199/199  | 100%       |
#### Where should the reviewer start?
`grommet/src/js/components/Select/__tests__/Select-test.js`
#### What testing has been done on this PR?
Run `yarn test Select-test` to confirm that all the tests pass
#### How should this be manually tested?
See above

#### What are the relevant issues?
#4219 
#### Screenshots (if appropriate)
Testing coverage before adding the new tests
Screen Shot 2020-06-25 at 6 09 13 PM
Testing coverage after
Screen Shot 2020-06-25 at 6 19 13 PM

#### Do the grommet docs need to be updated?
No
#### Should this PR be mentioned in the release notes?
No
#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

**Comment by JoeButler7** [comment]
### Update:
Added accessibility tests

File: src/js/components/Select/__tests__/Select-test.js [hunk 1]
```diff
@@ -14,6 +17,16 @@ describe('Select', () => {
 
   afterEach(cleanup);
 
+  test('should have not accessibility violations', async () => {
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/Select-test.js:20]:
```suggestion
  test('should not have accessibility violations', async () => {
```

File: src/js/components/Select/__tests__/Select-test.js [hunk 2]
```diff
@@ -61,8 +78,18 @@ describe('Select', () => {
       expect(document.activeElement).toMatchSnapshot();
       done();
     }, 100);
-  });
+    expect(onOpen).toHaveBeenCalled();
+    fireEvent.click(getByPlaceholderText('test select'));
+
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/Select-test.js:84]:
```suggestion
    // Verifing the Drop is closed once the placholder eas clicked on
    expect(container.firstChild).toMatchSnapshot();
```

File: src/js/components/Select/__tests__/Select-test.js [hunk 3]
```diff
@@ -61,8 +78,18 @@ describe('Select', () => {
...
+    expect(document.getElementById('test-select__drop')).toBeNull();
 
+    setTimeout(() => {
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/Select-test.js:87]:
```suggestion
    // Waiting for the drop to close
    setTimeout(() => {
```

File: src/js/components/Select/__tests__/Select-test.js (hunk 2 above)
Comment (ShimiSun) [reply src/js/components/Select/__tests__/Select-test.js:83]: Any reason we are not taking this commit?

Comment (JoeButler7) [reply src/js/components/Select/__tests__/Select-test.js:84]: In the most recent push I added comments throughout the tests and there is a comment here

**Comment by JoeButler7** [comment]
### Update:
Separated tests into `SelectBehavior-test.js` and `SelectRender-test.js`. `SelectBehavior-test.js` contains tests more focused on behavioral aspects of the select component such as selecting an option, deselecting an option, etc. `SelectRender-test.js` contains tests more focused on how the select component renders. The coverage was not impacted by separating the files

**Comment by JoeButler7** [comment]
### Update:
The files have been separated into `SelectControlled-test.js` for controlled inputs and `SelectUncontrolled-test.js` for uncontrolled inputs. Coverage was not impacted and now when running `yarn test-update` `SelectUncontrolled-test.js`finished in 8.971 seconds and `SelectControlled-test.js` finishes in 7.013 seconds, which is a significant reduction from the almost 40 seconds it took before the files were separated

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js [hunk 4]
```diff
@@ -10,10 +13,20 @@ import { Grommet } from '../..';
 import { Select } from '..';
 
 describe('Select', () => {
+  window.scrollTo = jest.fn();
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/SelectUncontrolled-test.js:16]: Why is this addition needed?

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js [hunk 5]
```diff
@@ -10,10 +13,20 @@ import { Grommet } from '../..';
...
   beforeEach(createPortal);
-
   afterEach(cleanup);
 
+  test('should not have accessibility violations', async () => {
+    const { container } = render(
+      <Grommet>
+        <Select options={['one', 'two', 'three']} a11yTitle="test" />
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/SelectUncontrolled-test.js:23]: we should add `a11yTitle` on the 'basic'  test below as well.

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js [hunk 6]
```diff
@@ -28,39 +41,40 @@ describe('Select', () => {
     expect(component.toJSON()).toMatchSnapshot();
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/SelectUncontrolled-test.js:41]: we can move this line to the first test of axe, and get rid of this 'a11yTitle' test since its testing are already included in the axe testing.

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js [hunk 7]
```diff
@@ -28,39 +41,40 @@ describe('Select', () => {
...
 
     fireEvent.click(getByPlaceholderText('test select'));
-
     expect(container.firstChild).toMatchSnapshot();
     expectPortal('test-select__drop').toMatchSnapshot();
+    // advance timers so the select opens
+    jest.advanceTimersByTime(100);
+    // verify that select is open
+    expect(document.activeElement).toMatchSnapshot();
+    expect(onOpen).toHaveBeenCalled();
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/SelectUncontrolled-test.js:67]: Let's avoid `toHaveBeenCalled` and use `toHaveBeenCalledTimes` instead for more accuracy. true for the rest of the PR.

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js (hunk 4 above)
Comment (JoeButler7) [reply src/js/components/Select/__tests__/SelectUncontrolled-test.js:16]: Because every time the select is clicked on  `window.scrollTo` is called and if it is not mocked it gives an error. Since the component is clicked on in almost every test the cleanest way to implement it was to mock the function for the entire file then restore it at the end of the file. 
Screen Shot 2020-07-08 at 12 35 17 PM

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js (hunk 7 above)
Comment (JoeButler7) [reply src/js/components/Select/__tests__/SelectUncontrolled-test.js]: Changed in the most recent push

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js (hunk 5 above)
Comment (JoeButler7) [reply src/js/components/Select/__tests__/SelectUncontrolled-test.js:23]: Included in the most recent push

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js (hunk 6 above)
Comment (JoeButler7) [reply src/js/components/Select/__tests__/SelectUncontrolled-test.js:35]: Has been changed in the most recent push

File: src/js/components/Select/__tests__/SelectUncontrolled-test.js [hunk 8]
```diff
@@ -58,6 +58,10 @@ describe('Select', () => {
     jest.advanceTimersByTime(100);
     // verify that select is open
     expect(document.activeElement).toMatchSnapshot();
+    /* called twice because of bug, issue #4283
+    https://github.com/grommet/grommet/issues/4283
+    Test should be changed when bug is fixed
```
Comment (ShimiSun) [inline src/js/components/Select/__tests__/SelectUncontrolled-test.js:63]:
```suggestion
    Test should be changed to 'toHaveBeenCalledTimes(1)' when the bug is fixed
```

**Comment by ShimiSun** [comment]
linking to issue #4283

**Comment by JoeButler7** [comment]
### Update:
Separated tests by multiple and non-multiple into the files `SelectMultiple-test.js` and `Select-test.js` Coverage was not impacted

**Comment by ShimiSun** [comment]
Great work, great coverage!! 💟
